tensorflow/python/framework/device_test.py

```diff
@@ -82,11 +82,9 @@ def testCheckValid(self):
 
     with self.assertRaisesRegex(ValueError, "invalid literal for int"):
       device.check_valid("/job:j/replica:foo")
-
     with self.assertRaisesRegex(ValueError, "invalid literal for int"):
       device.check_valid("/job:j/task:bar")
-
-    with self.assertRaisesRegex(ValueError, "Unknown attribute: 'bar'"):
```

Comment:
These seem like unintended changes. Please revert them back.

Reply:
These changes are intended. "/bar:muu/baz:2", originally, this test will check the /bar is an invalid device name, and reported "Unknown attribute : `bar`", with PluggableDevice enabled, device type checking is removed from device_spec.py, and the "bar" can be a device name, but "muu" is not a valid device Id, so I change the test from "Invalid attribute `bar`" -> "invalid literal for int"

Reply:
Please keep this test. To make sure the test works correctly, let's first check with `tf.config.list_physical_devices()` whether there is a device type `bar` or `baz`. If there is, come up with something that isn't in the list. (If not, we can keep using `bar` and `baz`.) Then we can perform the check as usual.

Reply:
This test is a negative case(intend to test the device_type is invalid) , so the device.check_valid() may return true if plug in a bar/baz device type or it return false. With self.assertRaiseRegex will expect that it will report Unknown attribute : 'bar', so the test may pass(didn't register a bar/baz device) and may fail(dynamically registered a bar/baz device).

So seems it is hard to construct a case that will always report an invalid device type, do we need to name it as something like NotADevice and expect no one will register this? Thanks

Reply:
How about using `bar` followed by all PluggableDevice type names? E.g,
```Python
# `list_of_pluggable_device_type_names()` parses the results from 
# `config.list_physical_devices()` and returns a list of all device type names
# For example, if there are PluggableDevices ["XPU", "VPU"], then the name
# would be `barXPUVPU`.
device_name_to_check = 'bar' + ''.join(list_of_pluggable_device_type_names())
``` 
This one can't be in the list. And if there's no PluggableDevice, the name is just `bar` as usual.